python_modules/dagster/dagster/_core/execution/asset_backfill.py

```diff
@@ -56,7 +56,7 @@ class AssetBackfillData(NamedTuple):
     def is_complete(self) -> bool:
         return (
             (
-                self.requested_subset | self.failed_and_downstream_subset
+                self.materialized_subset | self.failed_and_downstream_subset
```

Comment:
If I understand correctly, a consequence of this change is that the record for this backfill in the bulk actions table will no longer be marked as COMPLETE until all the targeted asset partitions have either been materialized, failed, or skipped.

This means that the meaning of COMPLETE in the bulk actions table will be different for asset backfills than for job backfills.

@prha do you have thoughts on whether this will have consequences? The alternative would be to introduce a new bulk action status that encodes that all the work requested on behalf of the bulk action is complete. I'm having trouble coming up with a good name for this, but haven't thought it too hard about it yet.

Reply:
Backfill status is an overloaded term.

The original intent of this column is purely to track state on the behalf of the backfill daemon.  The backfill daemon has completed its work for a given backfill job if it has requested a run for every partition.  It can then filter by this column to find more work that it should do.

This is completely separate from the user-facing concept of backfill status (i.e. have all the runs completed successfully for a backfill).  We make sure the UI reflects the individual run status rather than just reading from the backfill status column.

For asset backfills, even though the particulars are different, we should still maintain this split between the concept of "complete" from the daemon perspective (all runs requested), and from the user perspective (backfill is done).

I'm not as up to speed on the asset backfill particulars (the logic is quite complicated and involves the cache queryer).  If a run fails, will the backfill retry?  If not, then I think "complete" for the daemon means that all runs are requested.  That ensures that we won't request duplicate runs on the next backfill iteration.

Reply:
> If not, then I think "complete" for the daemon means that all runs are requested. 

The potential issue here is that this PR introduces a distinction between:
- "We no longer need to request any more runs for this backfill"
- "We no longer need to process this backfill within backfill iterations"

I.e., with this PR, we want to keep loading asset backfills after we're done requesting runs, in order to update the `AssetBackfillData` object. This object already needs to keep track of the final status (materialized/failed) of some of the asset partitions in the backfill, so if we can make `AssetBackfillData` keep track of _all_ of them, it makes it simpler and cheaper to query for info about the backfill's progress for the UI.

This means that we need some status that can allow us to distinguish between:
- Backfills where all the requested runs have completed, and thus we should never even load them in the backfill daemon
- Backfills that we want to load in the backfill daemon, either to request new runs for or just to update the data we're tracking for them

> I'm not as up to speed on the asset backfill particulars (the logic is quite complicated and involves the cache queryer). If a run fails, will the backfill retry? 

No

Reply:
> Backfills where all the requested runs have completed, and thus we should never even load them in the backfill daemon
> Backfills that we want to load in the backfill daemon, either to request new runs for or just to update the data we're tracking for them

As long as the backfill daemon can distinguish between the two (and avoid rerequesting runs unnecessarily), I'm fine with `BulkActionStatus.COMPLETE` to represent the latter.  It's not clear to me that having another state to keep track of will help avoid future confusion.

Hopefully we can entomb some of this discussion in comments in this PR.